core/shared/src/main/scala/zio/Clock.scala

```diff
@@ -265,147 +137,40 @@ object Clock extends ClockPlatformSpecific with Serializable {
 
   }
 
-  // Accessors
-
   /**
    * Returns the current time, relative to the Unix epoch.
    */
-  def currentTime(unit: => TimeUnit)(implicit trace: ZTraceElement): URIO[Clock, Long] =
-    ZIO.serviceWithZIO(_.currentTime(unit))
+  def currentTime(unit: => TimeUnit)(implicit trace: ZTraceElement): UIO[Long] =
+    ZIO.runtimeConfig.flatMap(_.services.get[Clock].currentTime(unit))
 
   /**
    * Get the current time, represented in the current timezone.
    */
-  def currentDateTime(implicit trace: ZTraceElement): URIO[Clock, OffsetDateTime] =
-    ZIO.serviceWithZIO(_.currentDateTime)
-
-  def driver[Env, In, Out](
-    schedule: Schedule[Env, In, Out]
-  )(implicit trace: ZTraceElement): URIO[Clock, Schedule.Driver[schedule.State, Env, In, Out]] =
-    ZIO.serviceWithZIO(_.driver(schedule))
+  def currentDateTime(implicit trace: ZTraceElement): UIO[OffsetDateTime] =
+    ZIO.runtimeConfig.flatMap(_.services.get[Clock].currentDateTime)
 
-  def instant(implicit trace: ZTraceElement): ZIO[Clock, Nothing, java.time.Instant] =
-    ZIO.serviceWithZIO(_.instant)
+  def instant(implicit trace: ZTraceElement): UIO[java.time.Instant] =
+    ZIO.runtimeConfig.flatMap(_.services.get[Clock].instant)
```

Comment:
Consider introducing a helper in each companion object, along the lines of:

```scala
def clockWith[R, E, A](f: Clock => ZIO[R, E, A]): ZIO[R, E, A] =
  ZIO.runtimeConfig.flatMap(config => f(config.services.get[Clock]))
```

This eliminates the duplication in writing accessors.

Reply:
Also, this could be abstracted a bit:

```scala
object ZIO {
  def standardWithZIO[Service <: Clock with System ...: Tag, R, E, A](f: Service => ZIO[R, E, A]): ZIO[R, E, A] = ???
}
```

Then `clockWith`, if you have it, is simply: `standardWithZIO[Clock](f)`.